toolchain/sem_ir/facet_type_info.h

```diff
@@ -105,14 +99,14 @@ struct IdentifiedFacetType {
 
   IdentifiedFacetType() {}
 
+  // The order here defines the order of impl witnesses for this facet type.
   auto required_interfaces() const -> llvm::ArrayRef<RequiredInterface> {
     return required_interfaces_;
   }
 
-  auto set_required_interfaces(const llvm::ArrayRef<RequiredInterface> set_to) {
-    required_interfaces_.assign(set_to.begin(), set_to.end());
-    CanonicalizeRequiredInterfaces();
-  }
+  auto set_required_interfaces(
```

Comment:
This is now setting all the fields of the type at once, right? Would it make sense to do this as a ctor instead? Or do we need to mutate?

```suggestion
  auto SetRequiredInterfaces(
```

If we keep it as a setter, it's no longer a simple field setter and should get a CamelCase name

Reply:
Made a constructor.